### PR TITLE
Ncnews 9

### PR DESCRIPTION
**Pull request by dmarkesini**
completed ticket 9

-used TDD
-error handling utilised for 400 and 404 errors

### CODE REVIEW

File: models/articles.models.js [hunk 1]
```diff
@@ -38,6 +38,20 @@ exports.updateArticleById = (inc_votes, id) => {
...
+exports.selectCommentsById = (id) => {
+  return db
+    .query(
+      `SELECT comment_id, votes, created_at, author, body FROM comments
+ WHERE article_id = ${id}`
+    )
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Comment not found!" });
+      }
```
Comment (cjplatten) [inline models/articles.models.js:56]: What will happen in the case that an article exists but doesn't have any comments associated with it?

**Review by cjplatten — COMMENTED** [review]
One comment to address

File: __tests__/app.test.js [hunk 2]
```diff
@@ -228,3 +227,53 @@ describe("GET /api/articles", () => {
...
+        expect(body.msg).toBe("Comment not found!");
+      });
+  });
+  test("status: 404 for an article_id that exists but has no comments", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Comment not found!");
+      });
```
Comment (cjplatten) [inline __tests__/app.test.js:279]: If the article exists but there are no comments this is not an error, the request has been successful it's just that the response is empty.

**Review by cjplatten — CHANGES_REQUESTED** [review]
One comment to address, think about how you'll handle the distinction between an empty query response due to the article not existing and an empty response due to the article existing but not having any comments.

**Review by karlaferry — APPROVED** [review]

**Pull request merged**